stablehlo/reference/ProcessGrid.h

```diff
@@ -31,31 +31,103 @@ limitations under the License.
 namespace mlir {
 namespace stablehlo {
 
+struct ProcessId;
+class RendezvousResult;
+
+namespace {
```

Comment:
Anonymous namespace in a header file is rarely a good idea. Creates a new namespace per translation unit.

https://stackoverflow.com/questions/357564/uses-for-anonymous-namespaces-in-header-files

I think keeping in ProcessGrid is OK, the alternative would be to add a `detail` namespace, which tells users that these things are implementation details of process grid and should not be relied upon, this is also an OK option. We use `detail` namespace in `Tensor.h`.

Reply:
I also thought about taking this out of ProcessGrid to something like ThreadSafeUtils.h. These are generic enough to be separated, but this is also exclusively used by ProcessGrid, and I don't anticipate any other uses of these classes. Would be good to add `detail` to be consistent for now though.

Reply:
Detail is fine. Prefer to not make things overly generic/reusable until there is a need for it